From 786b46fc419b2a8b09d85181313bac9d18cb8f26 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 11 Jun 2025 09:23:45 +0200
Subject: [PATCH 945/985] evl/net: ipv4: align on upstream changes to
 ip_route_output()

Since v6.10, the route lookup scope is given explicitly to
ip_route_output().

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/net/ipv4/ipv4.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/evl/net/ipv4/ipv4.c b/kernel/evl/net/ipv4/ipv4.c
index 7219f297d..581353b4e 100644
--- a/kernel/evl/net/ipv4/ipv4.c
+++ b/kernel/evl/net/ipv4/ipv4.c
@@ -252,7 +252,7 @@ int evl_net_ipv4_solicit(struct evl_socket *esk,
 	if (flags & ~EVL_NEIGH_PERMANENT)
 		return -EINVAL;
 
-	rt = ip_route_output(sock_net(esk->sk), ipaddr, 0, RTO_ONLINK, 0);
+	rt = ip_route_output(sock_net(esk->sk), ipaddr, 0, 0, 0, RT_SCOPE_LINK);
 	if (IS_ERR(rt))
 		return PTR_ERR(rt);
 
-- 
2.30.2

